### PR TITLE
LPv2: Isolate gateway weights

### PR DESCRIPTION
**Pull request by lemunozm**
# Description

Small PR to move the weight checks to `MessageProcessor`.  

Like the fact that now only the `MessageProcessor` trait implementation knows about weights

### CODE REVIEW

File: pallets/liquidity-pools-gateway/src/lib.rs [hunk 1]
```diff
@@ -517,18 +517,31 @@ pub mod pallet {
...
+
+					let res = Self::process_inbound_message(
+						domain_address,
+						message,
+						router_id,
+						&mut counter,
+					);
+
+					let weight = match counter {
+						0 => LP_DEFENSIVE_WEIGHT / 10,
```
Comment (lemunozm) [inline pallets/liquidity-pools-gateway/src/lib.rs:531]: We want to add just a bit if we fail even before starting to process the first message

File: pallets/liquidity-pools-gateway/src/message_processing.rs [hunk 2]
```diff
@@ -437,27 +422,19 @@ impl<T: Config> Pallet<T> {
 				expected_proof_count,
 			);
 
-			if let Err(e) = inbound_entry.validate(&router_ids, &router_id.clone()) {
-				return (Err(e), weight.saturating_mul(count));
-			}
-
-			if let Err(e) = Self::upsert_pending_entry(message_proof, &router_id, inbound_entry) {
```
Comment (cdamian) [inline pallets/liquidity-pools-gateway/src/message_processing.rs:444]: My eyes are not bleeding anymore when looking at this, thank you <3

File: pallets/liquidity-pools-gateway/src/lib.rs (hunk 1 above)
Comment (cdamian) [reply pallets/liquidity-pools-gateway/src/lib.rs:531]: Should we maybe add a `LP_MIN_WEIGHT` or something?

File: pallets/liquidity-pools-gateway/src/message_processing.rs (hunk 2 above)
Comment (lemunozm) [reply pallets/liquidity-pools-gateway/src/message_processing.rs:444]: BTW, liked that you move the outbound processing impl into the `MessageProcessor`, now everything is quite unified 🙌🏻

File: pallets/liquidity-pools-gateway/src/lib.rs (hunk 1 above)
Comment (wischli) [reply pallets/liquidity-pools-gateway/src/lib.rs:531]: I would prefer hardcoding that as well, yes. An alternative would be a simple `fn defensive_weight(counter)`.

**Review by wischli — APPROVED** [review]
Thanks for tackling this! LGTM but I would prefer hardcoding the reduced defensive weight.

File: pallets/liquidity-pools-gateway/src/lib.rs (hunk 1 above)
Comment (lemunozm) [reply pallets/liquidity-pools-gateway/src/lib.rs:531]: I didn't read it, and I had the auto-merge enabled. Tech-debt

**Pull request merged**